ng-dev/misc/new-main-branch/remotes.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {GitClient} from '../../utils/git/git-client';
+import {debug} from '../../utils/console';
+
+/**
+ * Regular expression matching a remote verbose info line, capturing
+ * the remote name, remote url and type (for different mirrors).
+ *
+ * ```
+ *   origin  https://github.com/devversion/dev-infra.git (fetch)
+ *   origin  https://github.com/devversion/dev-infra.git (push)
+ *   upstream        https://github.com/angular/dev-infra.git (fetch)
+ *   upstream        https://github.com/angular/dev-infra.git (push)
+ * ```
+ */
+const remoteVerboseInfoRegex = /^([^\s]+)\s+([^\s]+)\s+\((fetch|push)\)$/;
```

Comment:
We need to also need to account for other characters in remote names as there you can do things like having a `-` in the name: [stack overflow](https://stackoverflow.com/questions/41461152/which-characters-are-illegal-within-a-git-remote-name)